source/developer/redux.rst

```diff
@@ -0,0 +1,261 @@
+
+If your selector needs to select based on some more advanced requirements or needs the result in a specific format then you'll need to make use of the ``createSelector`` function from `reselect <reselect <https://github.com/reactjs/reselect>`__. If you're not sure what this is good for, `check out the previously mentioned developer talk <https://www.youtube.com/watch?v=6N2X7gEwmaQ>`__. The short form reason is using reselect allows for memoization and only runs the computation of selectors when the state affecting that selector has actually changed.
+
+The basic usage for ``createSelector`` is pass it all the selector functions needed as inputs to your computation. The last argument will then be a function that takes in the results of each selector you specified and performs your computation on that data before returning it.
+
+.. code-block:: javascript
+  :linenos:
+
+  export const getUsersByUsername = createSelector(
+      getUsers,
+      (users) => {
+          const usersByUsername = {};
+
+          for (const id in users) {
+              if (users.hasOwnProperty(id)) {
+                  const user = users[id];
+                  usersByUsername[user.username] = user;
+              }
+          }
+
+          return usersByUsername;
+      }
+  );
+
+Here we're using the ``getUsers`` selector to feed users into our function that builds a map of users with username as the key.
+
+So far that's pretty straight forward but what if you want to select some data based on an argument but also need to do some computation? That is a little more tricky if you haven't wrapped your head around the purpose of reselect and how createSelector works, so if you haven't watched the developer talk linked above I would strongly suggest it.
+
+To accomplish this we need to create factory function that will create the selector, instead of just creating the selector directly.
+
+.. code-block:: javascript
+  :linenos:
+
+  function getAllFiles(state) {
+      return state.entities.files.files;
+  }
+
+  function getFilesIdsByPosts(state, post) {
+      return state.entities.files.fileIdsByPostId;
+  }
+
+  export function makeGetFilesForPost() {
+      return createSelector(
+          getAllFiles,
+          getFilesIdsForPost,
+          (state, postId) => postId,
+          (allFiles, fileIdsForPost, postId) => {
+              return fileIdsForPost.map((id) => allFiles[id]);
+          }
+      );
+  }
+
+  // Usage by a third party application
+  const getFilesForPost = makeGetFilesForPost();
+  const files = getFilesForPost(state, 'somepostid');
+
+This can look a bit confusing but there is little happening here we haven't seen before. All that we're doing is using three selectors with ``createSelector``, the third selector just happens to be returning an argument so that our final function has access to it. Remember that every selector always takes state in as the first argument.
```

Comment:
"a bit confusing but there is" -> "a bit confusing, but there is"
"returning an argument" -> "returning its second argument"